.cargo/config.toml

```diff
@@ -1,5 +1,58 @@
-# Deployment runtime lints
-[target.'cfg(all(target_arch = "wasm32", not(test)))']
+# Unfortunately we cannot add any lints based on `cfg(test)` nor `cfg(feature = "feat")`: https://github.com/rust-lang/cargo/issues/8170.
+# Hence, we can only define lints that are applied codebase-wide (including tests), and lints that are applied only to runtime code.
+
+# Codebase-wide lints.
+[target.'cfg(all())']
+rustflags = [
+  # We need to deny warnings here instead of with the `RUSTFLAGS` env since the env variable would completely override these settings -> https://github.com/rust-lang/cargo/issues/5376.
+  "-Dwarnings",
+  "-Wclippy::as_underscore",
+  "-Wclippy::branches_sharing_code",
+  "-Wclippy::clear_with_drain",
+  "-Wclippy::clone_on_ref_ptr",
+  "-Wclippy::collection_is_never_read",
+  "-Wclippy::derive_partial_eq_without_eq",
+  "-Wclippy::else_if_without_else",
+  "-Wclippy::empty_drop",
+  "-Wclippy::empty_structs_with_brackets",
+  "-Wclippy::equatable_if_let",
+  "-Wclippy::if_then_some_else_none",
+  "-Wclippy::impl_trait_in_params",
+  "-Wclippy::iter_on_empty_collections",
+  "-Wclippy::iter_on_single_items",
+  "-Wclippy::iter_with_drain",
+  "-Wclippy::needless_collect",
+  "-Wclippy::needless_pass_by_ref_mut",
+  "-Wclippy::negative_feature_names",
+  "-Wclippy::option_if_let_else",
+  "-Wclippy::or_fun_call",
+  "-Wclippy::pub_without_shorthand",
+  "-Wclippy::redundant_clone",
+  "-Wclippy::redundant_type_annotations",
+  "-Wclippy::ref_patterns",
+  "-Wclippy::rest_pat_in_fully_bound_structs",
+  "-Wclippy::suspicious_operation_groupings",
+  "-Wclippy::type_repetition_in_bounds",
+  "-Wclippy::unnecessary_self_imports",
+  "-Wclippy::unnecessary_struct_initialization",
+  "-Wclippy::unneeded_field_pattern",
+  "-Wclippy::unused_peekable",
+  "-Wclippy::useless_let_if_seq",
+  "-Wclippy::wildcard_dependencies",
```

Comment:
For a better testing, we can also add: https://rust-lang.github.io/rust-clippy/master/index.html#assertions_on_result_states. 

Then we are enforced to check which error is thrown. We can still maintain the old behavior by assert_eq(result, Ok(_))

Reply:
I like this very much! https://github.com/KILTprotocol/kilt-node/pull/762/commits/a0e2f9b04f36224f880e194ff55cc1b574c956b8